openmeter/sink/storage.go

```diff
@@ -58,14 +59,30 @@ func (q InsertEventsQuery) ToSQL() (string, []interface{}, error) {
 
 	query := sqlbuilder.ClickHouse.NewInsertBuilder()
 	query.InsertInto(tableName)
-	query.Cols("namespace", "validation_error", "id", "type", "source", "subject", "time", "data")
+	query.Cols("namespace", "validation_error", "id", "type", "source", "subject", "time", "data", "ingested_at", "created_at")
 
 	for _, message := range q.Messages {
 		var eventErr string
 		if message.Status.Error != nil {
 			eventErr = message.Status.Error.Error()
 		}
 
+		ingestedAt := time.Now()
+		createdAt := time.Now()
+
+		for _, header := range message.KafkaMessage.Headers {
+			// Parse ingested_at header
+			if header.Key == "ingested_at" {
+				var err error
+
+				ingestedAtStr := string(header.Value)
+				ingestedAt, err = time.Parse(time.RFC3339, ingestedAtStr)
```

Comment:
```suggestion
				ingestedAt, err = time.Parse(time.RFC3339, string(header.Value))
```

Reply:
Note: this essentially resets `ingestedAt` to the zero value returned by `time.Parse`.